numpy/distutils/tests/test_system_info.py

```diff
@@ -285,3 +285,33 @@ def test_overrides(self):
         finally:
             os.chdir(previousDir)
         
+
+def test_distutils_parse_env_order():
+    from numpy.distutils.system_info import _parse_env_order
+    env = 'NPY_TESTS_DISTUTILS_PARSE_ENV_ORDER'
+
+    base_order = list('abcdef')
+
+    os.environ[env] = 'b,i,e,f'
+    order, unknown = _parse_env_order(base_order, env)
+    assert len(order) == 3
+    assert order == list('bef')
+    assert len(unknown) == 1
+
+    for prefix in '^!':
+        os.environ[env] = f'{prefix}b,i,e'
+        order, unknown = _parse_env_order(base_order, env)
+        assert len(order) == 4
+        assert order == list('acdf')
+        assert len(unknown) == 1
+
+    with pytest.raises(ValueError):
+        os.environ[env] = 'b,^e,i'
+        _parse_env_order(base_order, env)
+
+    with pytest.raises(ValueError):
+        os.environ[env] = '!b,^e,i'
+        _parse_env_order(base_order, env)
+
+    # clean-up environment
+    del os.environ[env]
```

Comment:
I think it would be more robust to use [pytest's `monkeypatch.setenv()`](https://docs.pytest.org/en/stable/monkeypatch.html) instead of relying on `del` to reset the environment

Reply:
Thanks! I thought I had scoured the `pytest` for this functionality since I needed it elsewhere, must have overlooked it!! Thanks! I'll adapt!